ui/src/dashboards/components/CellEditorOverlay.tsx

```diff
@@ -514,23 +524,44 @@ class CellEditorOverlay extends Component<Props, State> {
     return result
   }
 
-  private get sourceLink(): string {
+  private get initialSource(): Source {
     const {
       cell: {queries},
-      source: {links},
+      source,
+      sources,
     } = this.props
-    return _.get(queries, '0.source.links.self', links.self)
+
+    const initialSourceLink: string = getNested<string>(
+      queries,
+      '0.source',
+      null
+    )
+
+    if (initialSourceLink) {
+      const initialSource = sources.find(
+        s => s.links.self === initialSourceLink
+      )
+
+      return initialSource
+    }
+    return source
   }
 
   private get source(): Source {
     const {source, sources} = this.props
-    const query = _.get(this.state.queriesWorkingDraft, 0, {sourceLink: null})
+    const query = _.get(this.state.queriesWorkingDraft, 0, {source: null})
 
-    if (!query.sourceLink) {
+    if (!query.source) {
       return source
     }
 
-    return sources.find(s => s.links.self === query.sourceLink) || source
+    return (
+      sources.find(
+        s =>
+          s.links.self ===
+          getNested<string | null>(query, 'source.links.self', null)
+      ) || source
+    )
```

Comment:
Small style thing here.  This would probably read better if it wasn't a multiline single line callback followed by an `or`.  Perhaps something like this?

```
const foo =  getNested<string | null>(query, 'source.links.self', null)
const bar = sources.find(s => s.links.self === foo)
if (bar) {
    return bar
}

return source
```

Reply:
👍